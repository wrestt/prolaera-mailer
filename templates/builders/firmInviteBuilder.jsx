import React from 'react';
import { Email, renderEmail } from 'react-html-email';
import Footer from '../../components/footer';
import GenericInfo from '../../components/generic/genericInfo';
import builderHeader from '../../components/header';
import SubHeader from '../../components/subHeader';
import css from '../templateCSS.js';

const firmInviteBuilder = async (completeUser, completeCompany, imageUrl) => {
  try {
    const headerText = <p>You've been added to {completeCompany.company_name} on Prolaera</p>;
    const bodyText = (
      <div>
        <p>Your account is waiting for you and there are only a couple quick steps to complete.</p>
        <ol>
          <li>Click the button below to register your account.</li>
          <li>On the registration page simply enter a password for your new account and click "Register".</li>
        </ol>
      </div>
    );
    const footerText = <p>That's it! You're ready to explore the new tools and resources you have access to.</p>;
    const Header = await builderHeader(imageUrl);
    return renderEmail(
      <Email title={"You've been added to " + `${completeCompany.company_name}` + 'on Prolaera'} headCSS={css}>
        <Header />
        <SubHeader subText={"You've been"} header={'Added'} />
        <GenericInfo header={headerText} body={bodyText} footer={footerText} email={completeUser.email} />
        <Footer />
      </Email>
    );
  } catch (error) {
    throw error;
  }
};

export default firmInviteBuilder;
