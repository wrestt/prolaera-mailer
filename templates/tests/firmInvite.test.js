import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';
import GenericInfo from '../../components/generic/GenericInfo';
import writeFile from '../../lib/writeFileHelper';
import firmInviteBuilder from '../builders/firmInviteBuilder';

Enzyme.configure({ adapter: new Adapter() });
const user = { email: 'test@email.com' };
const company = { company_name: 'Test CPA Firm' };
const headerText = <p>You've been added to {company.company_name} on Prolaera</p>;
const bodyText = (
  <div>
    <p>Your account is waiting for you and there are only a couple quick steps to complete.</p>
    <ol>
      <li>Click the button below to register your account.</li>
      <li>On the registration page simply enter a password for your new account and click "Register".</li>
    </ol>
  </div>
);
const footerText = <p>That's it! You're ready to explore the new tools and resources you have access to.</p>;

describe('Firm Invite component', () => {
  it('creates and checks snapshot of FirmInvite component html', async () => {
    let inviteHtml = renderer.create(
      <GenericInfo header={headerText} body={bodyText} footer={footerText} email={user.email} />
    );
    let inviteJson = inviteHtml.toJSON();
    expect(inviteJson).toMatchSnapshot();
  });
});

describe('create html test file', () => {
  it.skip('writes an html file to firmInviteTest.html', async () => {
    const firmInviteEmail = await firmInviteBuilder(
      user,
      company,
      'https://assets.prolaera.com/prolaeraLogo_fullText.png'
    );
    const saved = await writeFile(firmInviteEmail, 'firmInviteTest.html');
    expect(saved).toEqual(true);
  });
});
