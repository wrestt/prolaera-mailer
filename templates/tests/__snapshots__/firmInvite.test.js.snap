// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Firm Invite component creates and checks snapshot of FirmInvite component html 1`] = `
<div
  style={
    Object {
      "border": "1px solid lightgray",
      "borderRadius": "5px",
      "boxShadow": "1px 1px 8px darkgray",
      "display": "block",
      "fontFamily": "Arial, sans-serif",
      "marginBottom": "20px",
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": "500px",
      "padding": "20px",
      "textAlign": "left",
    }
  }
>
  <table
    align="center"
    border="0"
    cellPadding={0}
    cellSpacing={0}
    valign="top"
    width="100%"
  >
    <tbody>
      <tr>
        <td>
          <div>
            <p>
              You've been added to 
              Test CPA Firm
               on Prolaera
            </p>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            <div>
              <p>
                Your account is waiting for you and there are only a couple quick steps to complete.
              </p>
              <ol>
                <li>
                  Click the button below to register your account.
                </li>
                <li>
                  On the registration page simply enter a password for your new account and click "Register".
                </li>
              </ol>
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            <p>
              That's it! You're ready to explore the new tools and resources you have access to.
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
